runtime/dancebox/tests/integration_test.rs

```diff
@@ -633,6 +651,240 @@ fn test_authors_paras_inserted_a_posteriori_with_collators_already_assigned() {
+            assert!(current_author() == AccountId::from(BOB));
+
+            // Alice and Bob collate in our chain
+            let alice_id = get_aura_id_from_seed(&AccountId::from(ALICE).to_string());
+            let bob_id = get_aura_id_from_seed(&AccountId::from(BOB).to_string());
+
+            assert_eq!(authorities(), vec![alice_id, bob_id]);
+
+            assert_ok!(
+                Registrar::register(origin_of(ALICE.into()), 1001.into(), empty_genesis_data()),
+                ()
+            );
+            assert_ok!(
+                Registrar::mark_valid_for_collating(root_origin(), 1001.into()),
+                ()
+            );
+            // Purchase only enough credits for 1 session
+            let credits_1001 = dancebox_runtime::Period::get() * 2;
+            assert_ok!(ServicesPayment::purchase_credits(
+                origin_of(ALICE.into()),
+                1001.into(),
+                credits_1001,
+                None,
+            ));
+
+            // Assignment should happen after 2 sessions
+            run_to_session(1u32);
+            let assignment = CollatorAssignment::collator_container_chain();
+            assert!(assignment.container_chains.is_empty());
+            run_to_session(2u32);
+            // Charlie and Dave should be assigned to para 1001
+            let assignment = CollatorAssignment::collator_container_chain();
+            assert_eq!(
+                assignment.container_chains[&1001u32.into()],
+                vec![CHARLIE.into(), DAVE.into()]
+            );
+
+            // No credits are consumed if the container chain is not producing blocks
+            run_block();
+            let credits = pallet_services_payment::BlockProductionCredits::<Runtime>::get(
+                &ParaId::from(1001),
+            )
+            .unwrap_or_default();
+            assert_eq!(credits, credits_1001);
+
+            // Simulate block inclusion from container chain 1001
+            let mut sproof = ParaHeaderSproofBuilder::default();
+            let slot: u64 = 5;
+            let mut s = ParaHeaderSproofBuilderItem::default();
+            s.para_id = 1001.into();
+            s.author_id = HeaderAs::NonEncoded(sp_runtime::generic::Header::<u32, BlakeTwo256> {
+                parent_hash: Default::default(),
+                number: 1,
+                state_root: Default::default(),
+                extrinsics_root: Default::default(),
+                digest: sp_runtime::generic::Digest {
+                    logs: vec![DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode())],
+                },
+            });
+            sproof.items.push(s);
+            // TODO: this seems to break because of the function `relay_chain_state_proof`, which is used to
+            // read randomness from the relay. Panic msg:
+            // 'Invalid relay chain state proof, already constructed in `set_validation_data`: RootMismatch'
+            set_author_noting_inherent_data(sproof);
```

Comment:
@girazoki need some help getting this to work, I am successfully mocking the container chain block inclusion, but later when running this line:

```
run_to_session(4u32);
```

it panics with

```
'Invalid relay chain state proof, already constructed in `set_validation_data`: RootMismatch'
```

Reply:
will take a look at this

Reply:
Done, thank you!